LibraBFT/Impl/Consensus/BlockStorage/BlockStore.agda

```diff
@@ -34,6 +48,44 @@ executeAndInsertBlockM b = do
       put (rmSetBlockStore s bs')
       ok eb
 
+executeAndInsertBlockE bs0 block =
+  maybeS (getBlock (block ^∙ bId) bs0) continue (pure ∘ (bs0 ,_))
+ where
+  continue : Either ErrLog (BlockStore _ × ExecutedBlock)
+  continue =
+    maybeS (bs0 ^∙ bsRoot _) (Left fakeErr) λ bsr →
+    let btRound = bsr ^∙ ebRound in
+    if-dec btRound ≥?ℕ block ^∙ bRound
+    then Left fakeErr -- block with old round
+    else do
+      eb ← case executeBlockE bs0 block of λ where
+        (Right res) → Right res
+        (Left (ErrBlockNotFound parentBlockId)) → do
+          eitherS (pathFromRoot parentBlockId bs0) Left $ λ blocksToReexecute →
+            case (forM) blocksToReexecute (executeBlockE bs0 ∘ (_^∙ ebBlock)) of λ where
+              (Left  e) → Left e
+              (Right _) → executeBlockE bs0 block
+        (Left err) → Left err
+      -- bs1 <- withErrCtx' (here []) (PersistentLivenessStorage.saveTreeE bs0 [eb^.ebBlock] [])
```

Comment:
Can these missing functions be postulated for now? Just so that the shape of this function is in a state ready for verification.

Reply:
E.g., `PersistentLivenessStorage.saveTreeE`

Reply:
DONE